### PR TITLE
Return quickly on OSX when 2M pages cannot be allocated

### PR DESCRIPTION
**Pull request by pshipton**
Attempting to allocate 2M pages with a memory range would take a very long time before failing. 

Fixed up the code a little at the same time. The changes don't look so big if you ignore whitespace.

### CODE REVIEW

**Comment by pshipton** [comment]
@0xdaryl is there anyone who can look at this?

**Comment by pshipton** [comment]
@fjeremic I'm looking for a review on this one. Can you help?

**Comment by 0xdaryl** [comment]
@babsingh : would you mind helping with the review on this one please?

**Comment by 0xdaryl** [comment]
@genie-omr build osx

File: port/osx/omrvmem.c [hunk 1]
```diff
@@ -676,32 +676,43 @@ getMemoryInRange(struct OMRPortLibrary *portLibrary, struct J9PortVmemIdentifier
...
-					return NULL;
-				}
-				memoryPointer = NULL;
+	
+#if defined(VM_FLAGS_SUPERPAGE_SIZE_2MB)
+	if (PPG_vmem_pageSize[2] == pageSize) {
+		/*
+		 * Don't bother scanning the entire address space if the request can't be satisfied.
+		 */
+		void *tmpPointer = reserveMemory(portLibrary, NULL, byteAmount, identifier, mode, pageSize, pageFlags, category);
```
Comment (babsingh) [inline port/osx/omrvmem.c:685]: Note: This is the new code. The remaining is reformatted code. Passing `NULL (= address)` to `reserveMemory` triggers an `mmap` with `NULL (= address)`. Here, the system tries to automatically select the `address`. If `mmap` fails, then there is no memory available. Assuming `mmap`'s fail path is quick, this should improve perf when no memory is available.

Which benchmark was used to verify the perf improvement?

**Review by babsingh — COMMENTED** [review]
The code logic looks good. Minor formatting changes suggested.

File: port/osx/omrvmem.c (hunk 1 above)
Comment (pshipton) [reply port/osx/omrvmem.c:685]: On a Mac if you run `java -Xlp:objectheap:pagesize=2m` it takes a really really long time to fail to allocate 2m and start java.

**Comment by pshipton** [comment]
@babsingh I'm made the suggested formatting changes.

File: port/osx/omrvmem.c (hunk 1 above)
Comment (pshipton) [reply port/osx/omrvmem.c:683]: Not the Windows omvvmem.c does a similar check.

File: port/osx/omrvmem.c [hunk 2]
```diff
@@ -676,32 +676,42 @@ getMemoryInRange(struct OMRPortLibrary *portLibrary, struct J9PortVmemIdentifier
...
+	while ((startAddress <= currentAddress) && (endAddress >= currentAddress)) {
+		memoryPointer = reserveMemory(portLibrary, currentAddress, byteAmount, identifier, mode, pageSize, pageFlags, category);
+
+		if (NULL != memoryPointer) {
+			/* stop if returned pointer is within range */
+			if ((startAddress <= memoryPointer) && (endAddress >= memoryPointer)) {
+				break;
+			}
+			if (0 != omrvmem_free_memory(portLibrary, memoryPointer, byteAmount, identifier)) {
+				goto done;
```
Comment (babsingh) [inline port/osx/omrvmem.c:701]: I think `memoryPointer` can be non-null before the `goto`. So, it should be set to `NULL` before the `goto` in order to match the old code.
```suggestion
				memoryPointer = NULL;
				goto done;
```

Comment (pshipton) [reply port/osx/omrvmem.c:702]: Right, thanks. Fixed.

**Review by babsingh — APPROVED** [review]
LGTM.

**Review by fjeremic — APPROVED** [review]

**Pull request merged**